test/models/summary_helper_test.exs

```diff
@@ -0,0 +1,390 @@
+defmodule CoursePlanner.SummaryHelperTest do
+  use CoursePlanner.ModelCase
+
+  import CoursePlanner.Factory
+  alias CoursePlanner.SummaryHelper
+
+  @empty_summary_helper_user_data_response %{terms: [], offered_courses: []}
+
+  test "when user role is unknown" do
+    student = insert(:student)
+    [term1, term2] = insert_list(2, :term)
+    insert_list(4, :offered_course, term: term1)
+    insert(:offered_course, term: term2, students: [student])
+
+    student_data = SummaryHelper.get_term_offered_course_for_user(student.id, "Unknown")
+    assert student_data == @empty_summary_helper_user_data_response
+  end
+
+  describe "Student summary data" do
+    test "when she does not exist" do
+      student_data = SummaryHelper.get_term_offered_course_for_user(-1, "Student")
+      assert student_data == @empty_summary_helper_user_data_response
+    end
+
+    test "when she is not registered to any offered_course" do
+      student = insert(:student)
+      insert(:class)
+
+      student_data = SummaryHelper.get_term_offered_course_for_user(student.id, student.role)
+      assert student_data == @empty_summary_helper_user_data_response
+    end
+
+    test "when she has a course in one term" do
+      student = insert(:student)
+      [term1, term2] = insert_list(2, :term)
+
+      insert_list(4, :offered_course, term: term1)
+      offered_course =
+        insert(:offered_course, term: term2, students: [student])
+        |> Repo.preload(:classes)
+
+      student_summary_data_response =
+        %{@empty_summary_helper_user_data_response | terms: [term2], offered_courses: [offered_course]}
```

Comment:
so, you suggest of having two different assert ? or just simply create a new map ?